mlrun/model_monitoring/db/stores/__init__.py

```diff
@@ -95,24 +87,40 @@ def get_model_endpoint_store(
 
 def get_store_object(
     project: str,
-    access_key: str = None,
+    store_type: typing.Optional[str] = None,
     secret_provider: typing.Callable = None,
+    **kwargs,
 ) -> StoreBase:
     """
     Getting the DB target type based on mlrun.config.model_endpoint_monitoring.store_type.
 
     :param project:         The name of the project.
-    :param access_key:      Access key with permission to the DB table.
+    :param store_type:      The type of the store target. See mlrun.model_monitoring.db.stores.ObjectStoreFactory
+                            for available store types.
     :param secret_provider: An optional secret provider to get the connection string secret.
 
     :return: `StoreBase` object. Using this object, the user can apply different operations on the
              model monitoring record such as write, update, get and delete a model endpoint.
     """
 
+    store_connection_string = mlrun.model_monitoring.helpers.get_connection_string(
+        secret_provider=secret_provider
+    )
+
+    if store_connection_string and (
+        store_connection_string.startswith("mysql")
+        or store_connection_string.startswith("sqlite")
+    ):
+        store_type = mlrun.common.schemas.model_monitoring.ModelEndpointTarget.SQL
+        kwargs["store_connection_string"] = store_connection_string
+
+    # Set the default store type if no connection has been set
+    store_type = store_type or mlrun.mlconf.model_endpoint_monitoring.store_type
+
     # Get store type value from ObjectStoreFactory enum class
-    store_type = ObjectStoreFactory(mlrun.mlconf.model_endpoint_monitoring.store_type)
+    store_type = ObjectStoreFactory(store_type)
 
     # Convert into store target object
     return store_type.to_object_store(
-        project=project, access_key=access_key, secret_provider=secret_provider
+        project=project, secret_provider=secret_provider, **kwargs
     )
```

Comment:
If I follow correctly, it fails if there's a connection string in the secrets, and we try to get a V3IO-based store.

Reply:
but if you have connection string, we update the `store_type` accordingly

Reply:
I see your point.
Generally, we would like to minimize surprises in the code.
If one called
```py
v3io_store = get_store_object(..., store_type="v3io-nosql", secret_provider=...)
```
They would be surprised to get a MySQL store instead of a V3IO KV store - and that situation might happen here.

So, I think the `store_type` argument should determine the returned type if explicitly provided.

Reply:
Actually we don't really need this param here. I mean we shouldn't pass it to `get_store_object` because anyway it should be decided based on the connection string (otherwise take it from the default config). Therefore I removed it. 
https://github.com/mlrun/mlrun/pull/5737/commits/922ee45ce586395d45ca97c58d8daadbfb667d6a#diff-7cf9046981c4e232dd7484cd0651eb603ecc22a53310b05e4d737bb190399df6:~:text=%3A%20str%2C-,store_type,-%3A%20typing.